src/components/mc/cpu/mc_cpu_reduce.h

```diff
@@ -9,77 +9,158 @@
 
 #include "utils/ucc_math.h"
 
-#define DO_DT_REDUCE_WITH_OP(s1, s2, d, count, OP) do {                        \
+#define OP_1(_s1, _s2, _i, _sc, _OP) _OP(_s1[_i], _s2[_i])
```

Comment:
Minor, manually substituting OP reduces mc_cpu compilation time from 68 sec. to 58 sec.

```cpp
#define OP_1(_s1, _s2, _i, _sc, _OP) _OP(_s1[_i], _s2[_i])
#define OP_2(_s1, _s2, _i, _sc, _OP) _OP(_OP(_s1[_i], _s2[_i]),  _s2[_i + 1 * _sc])
#define OP_3(_s1, _s2, _i, _sc, _OP) _OP(_OP(_s1[_i], _s2[_i]),  _OP(_s2[_i + 1 * _sc],  _s2[_i + 2 * _sc]))
#define OP_4(_s1, _s2, _i, _sc, _OP) _OP(_OP(_OP(_s1[_i], _s2[_i]),  _OP(_s2[_i + 1 * _sc],  _s2[_i + 2 * _sc])), _s2[_i + 3 * _sc])
#define OP_5(_s1, _s2, _i, _sc, _OP) _OP(_OP(_OP(_s1[_i], _s2[_i]),  _OP(_s2[_i + 1 * _sc],  _s2[_i + 2 * _sc])), _OP(_s2[_i + 3 * _sc], _s2[_i + 4 * _sc]))
#define OP_6(_s1, _s2, _i, _sc, _OP) _OP(_OP(_OP(_s1[_i], _s2[_i]),  _OP(_s2[_i + 1 * _sc],  _s2[_i + 2 * _sc])), _OP(_OP(_s2[_i + 3 * _sc], _s2[_i + 4 * _sc]),_s2[_i + 5 * _sc]))
#define OP_7(_s1, _s2, _i, _sc, _OP) _OP(_OP(_OP(_OP(_s1[_i], _s2[_i]),  _OP(_s2[_i + 1 * _sc],  _s2[_i + 2 * _sc])), _OP(_OP(_s2[_i + 3 * _sc], _s2[_i + 4 * _sc]),_s2[_i + 5 * _sc])), _s2[_i + 6 * _sc])
```

Reply:
After another round of hesitations i split reductions into several independent functions and put them into different .c files. The compilation now 4 secs. The interfaces of the MC is kept the same (it 1 reduce_multi function). @Sergei-Lebedev plz have another look